tools/cockpit.spec

```diff
@@ -257,12 +278,21 @@ rm -f %{buildroot}/%{_prefix}/share/metainfo/org.cockpit-project.cockpit-docker.
 
 sed -i "s|%{buildroot}||" *.list
 
+%if 0%{?suse_version}
+# remove other brandings as they contain stale symlinks
+pushd %{buildroot}/%{_datadir}/cockpit/branding
+ls -1 | grep -v default | xargs rm -vr
+popd
+# need this in SUSE as some post build check dislikes stale symlinks
+install -m 644 -D /dev/null %{buildroot}/run/cockpit/motd
+%else
```

Comment:
Please keep these steps (branding removal, /dev/null link, and disabling of dbg package) as independent paragraphs/if conditions. It's otherwise all too easy to accidentally damage this.

I wondered here whether your package scripts don't complain about the unpackaged file in `%{buildroot}/run`, but I figure that's achieved by the `%ghost` thing further down? (I've never seen that)

Reply:
ok so that one is used by pam_motd. We do not use that by default in openSUSE. login itself does not support reading the directory. So I'd simply disable that for openSUSE.
Also, newer pam_motd actually support /usr/lib/motd.d/, so putting the symlink there would be the better choice.
Similarly for issue.d. Agetty in util-linux 2.35 now also reads /usr/lib/issue.d, so that symlink can be moved as well.

Reply:
Right, I think we used /etc/motd.d/ back then because some of our supported OSes didn't support /usr/lib/motd.d/ yet. @allisonkarlitskaya  ? But indeed we should move that now, this will also make it easier to disable the message by creating a /etc/motd.d/cockpit → /dev/null message.

But that won't actually help with *this particular* issue, right? We still need the indirection via /run either way.

Reply:
hmpf, motd and issue link to the same file, so can't remove it

Reply:
ok so there is a way to skip the stale symlink check by setting NO_BRP_STALE_LINK_ERROR.  That would then of course hide legitimate errors in case there would be any in the future.
With the stale symlink check we need the %ghost /run entries in the file list (or %exclude), otherwise rpm complains about unpackaged files. What's your preference?

Reply:
Checking for stale symlinks and explicitly whitelist them with `%ghost` sounds very reasonable to me. Apparently Fedora's/RHEL's rpm build don't complain, but that doesn't mean that this is the right thing to do. :-)

Reply:
> Right, I think we used /etc/motd.d/ back then because some of our supported OSes didn't support /usr/lib/motd.d/ yet. @allisonkarlitskaya ? But indeed we should move that now, this will also make it easier to disable the message by creating a /etc/motd.d/cockpit → /dev/null message.

I think the reason I did it this way is because it was an easier sell to the PAM people.  Then distros just used the default, and here we are...